core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -517,6 +517,13 @@ class KafkaController(val config: KafkaConfig, zkClient: KafkaZkClient, time: Ti
    * elector
    */
   def startup() = {
+    zkClient.registerStateChangeHandler(new StateChangeHandler {
+      override val name: String = KafkaController.getClass.getName
+      override def onReconnectionTimeout(): Unit = error("Reconnection timeout.")
+      override def afterInitializingSession(): Unit = newSession()
+      override def onAuthFailure(): Unit = error("Auth failure.")
```

Comment:
We probably can just change the info level logging for auth failure in ZooKeeperClientWatcher to error and avoid doing the same logging here. Ditto in ZkStateChangeListener.